### PR TITLE
Use post table when Encoding has only Differences

### LINKED ISSUE #13107: Attached file isn't rendered properly
Attach (recommended) or Link to PDF file here:
[NAPEN16171380959.pdf](https://github.com/mozilla/pdf.js/files/6143802/NAPEN16171380959.pdf)

Configuration:
- Web browser and its version: Firefox 88 Nightly
- Operating system and its version: Windows 10
- PDF.js version:
- Is a browser extension: No

What went wrong? (add screenshot)

### PR DESCRIPTION
**Pull request by janpe2**
Fixes #13107
In the issue, some TrueType glyph names have the format `uniXXXX`.
Font's `Encoding` dictionary has the entry `Differences` but no
`BaseEncoding`. `uniXXXX` names are converted to glyph indices
using font's `post` table but currently that is done only when
`BaseEncoding` exists. We must enable the conversion also when only
`Differences` exists.

### CODE REVIEW

File: src/core/fonts.js [hunk 1]
```diff
@@ -2901,13 +2901,18 @@ var Font = (function FontClosure() {
...
-              const glyphId = properties.glyphNames.indexOf(glyphName);
-              if (glyphId > 0 && hasGlyph(glyphId)) {
-                charCodeToGlyphId[i] = glyphId;
+            if (charCodeToGlyphId[i] === undefined) {
+              glyphName = this.differences[i] || baseEncoding[i];
+              if (glyphName) {
+                const glyphId = properties.glyphNames.indexOf(glyphName);
+                if (glyphId > 0 && hasGlyph(glyphId)) {
+                  charCodeToGlyphId[i] = glyphId;
+                }
```
Comment (Snuffleupagus) [inline src/core/fonts.js:2915]: *Optional nit:* I'd be sort-of inclined to reduce the overall indentation level in this loop, by using `continue`, e.g. something like:
```javascript
            if (charCodeToGlyphId[i] !== undefined) {
              continue;
            }
            glyphName = this.differences[i] || baseEncoding[i];
            if (!glyphName) {
              continue;
            }
            const glyphId = properties.glyphNames.indexOf(glyphName);
            if (glyphId > 0 && hasGlyph(glyphId)) {
              charCodeToGlyphId[i] = glyphId;
            }
```

**Review by Snuffleupagus — CHANGES_REQUESTED** [review]
(Removed)

**Review by Snuffleupagus — APPROVED** [review]
r=me, with passing tests; thank you!

File: src/core/fonts.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (janpe2) [reply src/core/fonts.js]: Yes, less indentation is easier to read.

**Review by timvandermeij — APPROVED** [review]

**Pull request merged**